htmresearch/frameworks/layers/l2_l4_network_creation.py

```diff
@@ -303,12 +303,21 @@ def createMultipleL4L2Columns(network, networkConfig):
   for i in range(networkConfig["numCorticalColumns"]):
     suffixSrc = "_" + str(i)
     for j in range(networkConfig["numCorticalColumns"]):
-      if i != j:
-        suffixDest = "_" + str(j)
+      suffixDest = "_" + str(j)
+      if i < j:
+        # use delayed link for lateral connections because c_i is computed
+        # before c_j
         network.link(
-            "L2Column" + suffixSrc, "L2Column" + suffixDest,
-            "UniformLink", "",
-            srcOutput="feedForwardOutput", destInput="lateralInput")
+          "L2Column" + suffixSrc, "L2Column" + suffixDest,
+          "UniformLink", "",
+          srcOutput="feedForwardOutput", destInput="lateralInput",
+          propagationDelay=1)
+      elif i > j:
+        network.link(
+          "L2Column" + suffixSrc, "L2Column" + suffixDest,
+          "UniformLink", "",
+          srcOutput="feedForwardOutput", destInput="lateralInput",
+          propagationDelay=0)
```

Comment:
I created an issue to track this [here](https://jira.numenta.com/browse/NUP-2328). Not necessarily Vitaly that will work it but I will try to get it into the next sprint.